### ISSUE TITLE
Windows: ModernUi (App) support

### ISSUE THREAD

**Issue by Ghabry** [EasyRPG/Player#730]
Is now in development, TODO:
 - [x] Video output (Direct2D) 
 - [ ] Flicker free video out
 - [ ] Correct aspect ratio
 - [ ] Audio (XAudio2), needs GenericAudio)
 - [ ] Input (Keyboard)
 - [ ] Input (Gamepad)
 - [ ] Input (Touch UI)
 - [ ] StorageFolder/FileBrowser API stuff and permissions
 - [ ] Launching games, needs FileSystem abstraction of #1085 
 - [ ] Some configuration (Android-like, at least for game folders)
 - [ ] Test on Windows Phone
 - [ ] Submit to Windows Store

Just a collection of requirements for Windows Modern UI (Apps running on Windows Mobile and "Modern Windows Apps")

From toolchain side all requirements are now implemented. With EasyRPG Player running on Windows RT the compatibility with ARM was also shown.

The biggest issue with ModernUi is the permission system. The way file access works is very special and does not support standard C/C++ IO. All directories, except private directories of the app, are blocked from reading.

The user must select folder he gives permissions on via a FolderPicker (at least Windows has a built-in one :D) and the App must store the token in a FutureAccessList.
This is very similiar to the new permission system of Android 6.

These tokens can only be used in the StorageFile and StorageFolder Apis and the whole Api is asynchronous(!).
This means the whole FileIO must be rewritten (from FILE\* to a wrapper class).

Reusing parts of the FileRequestAsync Api should be possible. The only places where async really matters are graphic and sound. For sound we don't control the API, this needs investigation in SDL_mixer but the graphic part should be easy.

So in short: File access must be changed (again) but most of the difficult work was already achieved for emscripten. Rest should "just work".

**Comment by Ghabry** [comment]
When I read the docs correctly you can do synchronous read operations on the stream returned by StorageFile->OpenSequentialReadAsync. But this needs real testing first, if I'm right.
This would make porting super easy (except replacing FILE* with a wrapper) because we could recycle most of the emscripten logic which is only async on (first) open \o/

**Comment by Ghabry** [comment]
Made some progress without depending on SDL. As a bonus: Working XAML UI layer.

![pic1](https://cloud.githubusercontent.com/assets/1331889/22661368/6b57cbec-eca5-11e6-87f2-d3973cde38ca.png)

![screenshot_20170206_193542](https://cloud.githubusercontent.com/assets/1331889/22661375/6f612f8a-eca5-11e6-853f-b40d2f0d0628.png)
